leasing/models/invoice.py

```diff
@@ -30,3 +42,155 @@ def create_reference_number(self):
+
+        return elements
+
+    def get_laske_parties(self):
+        elements = []
+
+        if self.billing_contact:
+            elements.append(self.billing_contact.as_laske_xml('OrderParty'))
+            elements.append(self.billing_contact.as_laske_xml('BillingParty1'))
+            elements.append(Contact().as_laske_xml('BillingParty2'))
+            elements.append(self.billing_contact.as_laske_xml('PayerParty'))
+        else:
+            elements.append(Contact().as_laske_xml('OrderParty'))
+            elements.append(Contact().as_laske_xml('BillingParty1'))
+            elements.append(Contact().as_laske_xml('BillingParty2'))
+            elements.append(Contact().as_laske_xml('PayerParty'))
+
+        return elements
+
+    def get_laske_line_item(self):
+        line_item = ElementTree.Element('LineItem')
+
+        ElementTree.SubElement(line_item, 'GroupingFactor')
+        material = ElementTree.SubElement(line_item, 'Material')
+        material.text = 'TODO'
+        ElementTree.SubElement(line_item, 'MaterialDescription')
+        quantity = ElementTree.SubElement(line_item, 'Quantity')
+        quantity.text = '1,00'
+        ElementTree.SubElement(line_item, 'Unit')
+        net_price = ElementTree.SubElement(line_item, 'NetPrice')
+        net_price.text = '{:.2f}'.format(self.amount).replace('.', ',')
+        line_text_l1 = ElementTree.SubElement(line_item, 'LineTextL1')
+
+        lease = self.tenants.first().lease
+        lease_tenants = set(lease.tenants.all())
+        invoice_tenants = set(self.tenants.all())
+
+        if lease_tenants.difference(invoice_tenants):
+            names = [str(tenant.contact) for tenant in lease_tenants.difference(invoice_tenants)]
+            line_text_l1.text = 'Muita vuokraajia : {}'.format(', '.join(names))
+
+        ElementTree.SubElement(line_item, 'LineTextL2')
+        ElementTree.SubElement(line_item, 'LineTextL3')
+        line_text_l4 = ElementTree.SubElement(line_item, 'LineTextL4')
+        line_text_l4.text = '   Maksun suorittaminen: Maksu on suoritettava viimeistään eräpäivänä.'
+        line_text_l5 = ElementTree.SubElement(line_item, 'LineTextL5')
+        line_text_l5.text = ' Eräpäivän jälkeen peritään korkolain mukainen viivästyskorko ja'.format()
+        line_text_l6 = ElementTree.SubElement(line_item, 'LineTextL6')
+        line_text_l6.text = ' mahdollisista perimistoimenpiteistä perimispalkkio.'.format()
+        ElementTree.SubElement(line_item, 'ProfitCenter')
+        order_item_number = ElementTree.SubElement(line_item, 'OrderItemNumber')
+        order_item_number.text = 'TODO'
+        ElementTree.SubElement(line_item, 'WBS_Element')
+        ElementTree.SubElement(line_item, 'FunctionalArea')
+        ElementTree.SubElement(line_item, 'BusinessEntity')
+        ElementTree.SubElement(line_item, 'Building')
+        ElementTree.SubElement(line_item, 'RentalObject')
+
+        return line_item
+
+    def as_laske_xml(self):
```

Comment:
IMHO the serialization code shouldn't be in the model. It would be better to have a separate Python module for the laske serialization.